### PR TITLE
#8543 Add backend checks to access documents

### LINKED ISSUE #8543: Add backend checks to access documents
### Problem Description
Related to the #898 epic we are adding backend checks (e.g. for contacts #8508) to make sure a user has the required user right to access or maniulate data.
For documents we are currently using a check in the UI based on the related entity (e.g. CASE_EDIT), but don't check anything in the backend.

### Proposed Change
Decide what's the best approach here. Do we need DOCUMENT_VIEW, DOCUMENT_EDIT, ... user rights?
Whenever a document is created/edited deleted we should probabaly also check the user right relevant for the related entity.

- [ ] Add backend checks for DocumentTemplateFacadeEjb -> DOCUMENT_TEMPLATE_MANAGEMENT

### PR DESCRIPTION
**Pull request by leventegal-she**
Fixes #8543

### CODE REVIEW

**Review by ChristopherRiedel — CHANGES_REQUESTED** [review]

File: sormas-backend/src/main/java/de/symeda/sormas/backend/document/DocumentStorageService.java [hunk 1]
```diff
@@ -85,11 +84,11 @@ private Path computeRelativePath(Document document) {
...
-		setAttribute(filePath, "Author", userService.getCurrentUser().getUserName());
-		setAttribute(filePath, "Display Name", document.getName());
-		setAttribute(filePath, "Type", document.getMimeType());
-		setAttribute(filePath, "Upload Date", DateFormatHelper.formatDate(document.getCreationDate()));
+//		setAttribute(filePath, "Document UUID", document.getUuid());
+//		setAttribute(filePath, "Author", userService.getCurrentUser().getUserName());
+//		setAttribute(filePath, "Display Name", document.getName());
+//		setAttribute(filePath, "Type", document.getMimeType());
+//		setAttribute(filePath, "Upload Date", DateFormatHelper.formatDate(document.getCreationDate()));
 	}
```
[the remark this answers is not in the document]
Comment (leventegal-she) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/document/DocumentStorageService.java]: It doesn't work on Mac but I've reverted it now

[the remark this answers is not in the document]
Comment (MartinWahnschaffe) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/document/DocumentStorageService.java]: This means setAttribute is not work on Mac in general or is it something specific that would make sense to fix?

**Comment by MartinWahnschaffe** [comment]
Apart from the existing comments, it looks very good to me.

**Review by ChristopherRiedel — APPROVED** [review]

File: sormas-backend/src/main/java/de/symeda/sormas/backend/document/DocumentStorageService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (MartinWahnschaffe) [reply sormas-backend/src/main/java/de/symeda/sormas/backend/document/DocumentStorageService.java]: Looks like JDK was not supporting user defined file attributes on OS X.
It was finally implemented with Java 17 (!): https://bugs.openjdk.org/browse/JDK-8030048

I'd suggest to add a try catch block into the setAttribute method for UnsupportedOperationException

**Pull request merged**